src/tike/opt.py

```diff
@@ -16,6 +16,83 @@
 randomizer = np.random.default_rng()
 
 
+def adagrad(d, v=None, eps=1e-6):
```

Comment:
What is this function for? I don't see any utilization of it.

Reply:
Sorry! I left in some commits from unrelated work.